### PR TITLE
feat: update send modal retire flow

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1190

- simplify send credit flow with retire option
- allow `AmountField` to have an `auxiliarLabel`
- change `MyEcoCredits` modals width to take scrollbar into account
- add link to recipient in send tx confirmation modal

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to https://deploy-preview-1300--regen-registry.netlify.app/ecocredits/dashboard
2. Click send action to check new retire modal flow

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/modal/FormModalTemplate.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ const useStyles = makeStyles(theme => ({
       padding: `${theme.spacing(10.75)} ${theme.spacing(7.5)} ${theme.spacing(
         12.5,
       )}`,
-      maxWidth: theme.spacing(140),
+      maxWidth: theme.spacing(148),
```
Comment (flagrede) [inline web-components/src/components/modal/FormModalTemplate.tsx:14]: Take into account scrollbar width to match Erika's designs.

**Comment by flagrede** [comment]
@erikalogie you can check the new send flow with retire option here https://deploy-preview-1300--regen-registry.netlify.app/ecocredits/dashboard

**Comment by erikalogie** [comment]
@flagrede this looks good. I think maybe we should just eliminate the "These credits will be tradable." paragraph as it has that info in the tooltip and it looks a bit funky having it disappear. Also, let's change the title from "Total amount of credits" to "Amount of credits".  https://www.figma.com/file/seUiQM2SPt78KJiDL0Ea8m/Basic-Ecocredits-Ledger-View-%2F-Marketplace?node-id=4899%3A156146

File: web-components/src/components/modal/FormModalTemplate.tsx (hunk 1 above)
Comment (wgwz) [reply web-components/src/components/modal/FormModalTemplate.tsx:14]: nit; should we set a constant here? i.e. `SCROLL_BAR_WIDTH`, and then say `140 + SCROLL_BAR_WIDTH`

**Review by wgwz — APPROVED** [review]

**Review by mhagel — APPROVED** [review]

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**